src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -260,7 +260,7 @@ public override void VisitBinaryExpression(BinaryExpressionSyntax node)
         public override void VisitDeclarationExpression(DeclarationExpressionSyntax node)
         {
             var argumentSyntax = (ArgumentSyntax)node?.Parent;
-            var argumentListSyntax = (BaseArgumentListSyntax)argumentSyntax?.Parent;
+            var argumentListSyntax = argumentSyntax.FirstAncestorOrSelf<BaseArgumentListSyntax>();
```

Comment:
That code works fine for this situation. Custom code for this error situation would do the same thing.

---
In reply to: [90172670](https://github.com/dotnet/roslyn/pull/15548#discussion_r90172670) [](ancestors = 90172670,90172205,90163033)